clippy_lints/src/utils/hir_utils.rs

```diff
@@ -607,8 +607,8 @@ impl<'a, 'tcx: 'a> SpanlessHash<'a, 'tcx> {
                 }
             },
             ExprKind::Err => {},
-            ExprKind::Use(ref e) => {
-                let c: fn(_) -> _ = ExprKind::Use;
+            ExprKind::DropTemps(ref e) => {
+                let c: fn(_) -> _ = ExprKind::DropTemps;
                 c.hash(&mut self.s);
```

Comment:
Huh, this should hash `mem::discriminant` (or thereabouts) like rustc does.
Preexisting though, better not to fix it in this PR.
cc @oli-obk @Manishearth